addons/rearm/functions/fnc_getAllTurrets.sqf

```diff
@@ -0,0 +1,28 @@
+/*
+ * Author: Tuupertunut
+ * Returns all turrets in a vehicle.
+ *
+ * BIS command "allTurrets" does not return the driver turret at the time of writing (2017-07-16).
+ * This function just adds driver turret to the array returned by "allTurrets".
+ *
+ * Arguments:
+ * 0: Vehicle <OBJECT>
+ *
+ * Return Value:
+ * Turret paths <ARRAY>
+ *
+ * Example:
+ * [vehicle] call ace_rearm_fnc_getAllTurrets
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_vehicle"];
+
+private _turrets = allTurrets _vehicle;
+
+// Adding the driver turret "[-1]".
+_turrets pushBack [-1];
```

Comment:
I don't like macros. 😦  They don't behave in the same elegant way as code should, as the space issue shows.

But yeah, even if I made a macro **or** used just pure code in the place of the function call, it wouldn't be simple. As @jonpas said, `[[-1]] + allTurrets _vehicle` is unnecessarily slow, so it would have to be `pushBack`. Problem is, the `pushBack` method takes al least 3 lines, as you have to first get the array and save it to a variable, then edit it with `pushBack` and then use it.